workspaces/github-discussions/plugins/search-backend-module-github-discussions/src/collators/GithubDiscussionsCollatorFactory.ts

```diff
@@ -98,6 +103,13 @@ export class GithubDiscussionsCollatorFactory
     this.repliesBatchSize = options.repliesBatchSize;
     this.org = options.org;
     this.repo = options.repo;
+    this.visibilityPermission = createPermission({
```

Comment:
This is tricky. Here are a couple of comments on how we can move forward with this feature:

First, we would need to define the permission in a `common` package. This is necessary for adopters to reference the permission in their permission policy.

Second, the permission is marked with a `resourceType`, which means it can accept conditional decisions (which are handled backend side). The problem with this is that there is no backend service associated with the collator. To address this, we have two options: either we remove `resourceType: 'github-discussions'`, in which case  we don't need a backend for the plugin, or we create a backend for authorizing the permissions.